provider/mem/mem.go

```diff
@@ -145,58 +121,41 @@ func (a *Alerts) GetPending() provider.AlertIterator {
 		}
 	}()
 
-	return provider.NewAlertIterator(ch, done, err)
-}
-
-func (a *Alerts) getPending() ([]*types.Alert, error) {
-	a.mtx.RLock()
-	defer a.mtx.RUnlock()
-
-	res := make([]*types.Alert, 0, len(a.alerts))
-
-	for _, alert := range a.alerts {
-		res = append(res, alert)
-	}
-
-	return res, nil
+	return provider.NewAlertIterator(ch, done, nil)
 }
 
 // Get returns the alert for a given fingerprint.
 func (a *Alerts) Get(fp model.Fingerprint) (*types.Alert, error) {
-	a.mtx.RLock()
-	defer a.mtx.RUnlock()
-
-	alert, ok := a.alerts[fp]
-	if !ok {
-		return nil, provider.ErrNotFound
-	}
-	return alert, nil
+	return a.alerts.Get(fp)
 }
 
 // Put adds the given alert to the set.
 func (a *Alerts) Put(alerts ...*types.Alert) error {
-	a.mtx.Lock()
-	defer a.mtx.Unlock()
 
 	for _, alert := range alerts {
 		fp := alert.Fingerprint()
 
-		if old, ok := a.alerts[fp]; ok {
+		if old, err := a.alerts.Get(fp); err == nil {
```

Comment:
What happens when `Get` returns an error?

Reply:
Right now, get only returns an error if the alert isn't found. This is me using knowledge of the concrete struct implementing this store interface -- if it were some sort of remote store then it could be a network error, for example, in which case we would .. probably ? .. want to continue as it is.

Reply:
> This is me using knowledge of the concrete struct implementing this store interface

That seems a bit implicit to me. How about just catching the `err` anyways or adding a comment?